packages/perseus/src/widgets/__tests__/expression-mobile.test.tsx

```diff
@@ -0,0 +1,350 @@
+        );
+
+        fireEvent.touchStart(input);
+
+        // waiting because `visibility` is animated
+        await waitFor(() =>
+            expect(screen.getByRole("button", {name: "1"})).toBeVisible(),
+        );
+
+        userEvent.click(screen.getByRole("tab", {name: "Operators"}));
+        expect(screen.getByRole("button", {name: "Equals sign"})).toBeVisible();
+    });
+
+    it("can navigate to geometry page", async () => {
+        render(<ConnectedRenderer />);
+
+        const input = screen.getByLabelText(
+            "Math input box Tap with one or two fingers to open keyboard",
+        );
+
+        fireEvent.touchStart(input);
+
+        // waiting because `visibility` is animated
+        await waitFor(() =>
+            expect(screen.getByRole("button", {name: "1"})).toBeVisible(),
+        );
+
+        userEvent.click(screen.getByRole("tab", {name: "Geometry"}));
+        expect(screen.getByRole("button", {name: "Sine"})).toBeVisible();
+    });
+
+    it("can navigate to extras page", async () => {
+        render(<ConnectedRenderer />);
+
+        const input = screen.getByLabelText(
+            "Math input box Tap with one or two fingers to open keyboard",
+        );
+
+        fireEvent.touchStart(input);
+
+        // waiting because `visibility` is animated
+        await waitFor(() =>
+            expect(screen.getByRole("button", {name: "1"})).toBeVisible(),
+        );
+
+        userEvent.click(screen.getByRole("tab", {name: "Extras"}));
+        expect(screen.getByRole("button", {name: "x"})).toBeVisible();
+    });
+
+    it("can write a full equation", async () => {
+        render(<ConnectedRenderer />);
+
+        const input = screen.getByLabelText(
+            "Math input box Tap with one or two fingers to open keyboard",
+        );
+
+        fireEvent.touchStart(input);
+
+        // waiting because `visibility` is animated
+        await waitFor(() =>
+            expect(screen.getByRole("button", {name: "1"})).toBeVisible(),
+        );
+
+        userEvent.click(screen.getByRole("tab", {name: "Extras"}));
+        userEvent.click(screen.getByRole("button", {name: "x"}));
+        userEvent.click(screen.getByRole("tab", {name: "Operators"}));
+        userEvent.click(screen.getByRole("button", {name: "Equals sign"}));
+        userEvent.click(screen.getByRole("tab", {name: "Geometry"}));
+        userEvent.click(screen.getByRole("button", {name: "Sine"}));
+        userEvent.click(screen.getByRole("tab", {name: "Numbers"}));
+        userEvent.click(screen.getByRole("button", {name: "9"}));
+
+        // MathQuill is problematic,
+        // this is the only way I know how to test the "input"
+        const mathquillInput =
+            // eslint-disable-next-line testing-library/no-node-access
+            document.getElementsByClassName("mq-root-block")[0] as HTMLElement;
+
+        expect(within(mathquillInput).getByText("x")).toBeVisible();
```

Comment:
NIT: Hmm does this test that the equation was typed correctly / in the correct order? Would this pass if it said sin()9=x? Totally an ignorable nit as I'm trying to think of things to provide commentary on, but is there a way to verify the equation as written?

Reply:
I wonder if we could use the HTML element to [get a handle](http://docs.mathquill.com/en/latest/Api_Methods/#mqhtml_element) to the Mathquill API and then [ask it](http://docs.mathquill.com/en/latest/Api_Methods/#latex) for the `.latex()` ... and use that to verify what's in it.  This test works, but feels clunky and potentially difficult to maintain.

Reply:
Updated